frontend/scripts/react-components/dashboard-element/dashboard-widget/table-modal/table/table.component.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'react-components/dashboard-element/dashboard-widget/table-modal/table/table.scss';
+import ShrinkingSpinner from 'react-components/shared/shrinking-spinner/shrinking-spinner.component';
+import { VariableSizeGrid } from 'react-window';
+import cx from 'classnames';
+import Text from 'react-components/shared/text';
+import maxBy from 'lodash/maxBy';
+
+const getMaxLength = row => String(maxBy(row, cell => String(cell).length)).length;
+
+function Table(props) {
+  const { data, headers, width, height, loading, rowHeight, className } = props;
+  if (!data) return null;
+  const columnWidth = width / headers.length;
+  const minRowHeight = 50;
+
+  return (
+    <table className={cx('c-table', className)}>
+      <thead>
+        <tr className="table-header" style={{ width }}>
+          {headers.map((h, index) => (
+            <th className="header-item" key={`${h}${index}`}>
+              <Text color="white" weight="bold" size="md" variant="mono">
+                {h}
+              </Text>
+            </th>
+          ))}
+        </tr>
+      </thead>
+      <VariableSizeGrid
+        className={cx('c-grid-list')}
+        height={height}
+        width={width}
+        innerElementType="tr"
+        columnWidth={() => columnWidth}
+        rowHeight={index => rowHeight || getMaxLength(data[index]) + minRowHeight}
+        rowCount={data.length}
+        columnCount={headers.length}
+      >
+        {({ rowIndex, columnIndex, style }) => {
+          const item = data[rowIndex][columnIndex];
+          if (typeof item === 'undefined') return null;
+          return (
+            <td
+              style={style}
+              className={cx('header-item', rowIndex % 2 ? 'list-item-odd' : 'list-item-even')}
```

Comment:
there a css pseudo-selector for this. https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child#Functional_notation

Reply:
We can't use it as we need the row index. React window displays directly the cells

Reply:
I see. I forgot about this 🤦‍♂️ . If we're putting all rows inside a single `<tr />` we lose all the table advantages. Doing some research I found that I was also wrong about the table element thing. https://github.com/bvaughn/react-window/issues/60

Apologies, I guess we shouldn't use a table then.

Reply:
Yeah, I also thought that about the tr. I will change it to divs and we will get rid of the warnings